File: day5/Factorial.java
import java.util.Scanner;

public class Factorial {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println(); // Blank line before
            System.out.print("Please enter a positive whole number: ");
            String userInput = scanner.nextLine();
            // No extra println() here

            try {
                int userNum = Integer.parseInt(userInput);

                if (userNum < 1) {
                    System.out.println();
                    System.out.println("Please only enter a positive whole number.");
                    continue;
                }

                long factorial = 1;
                for (int i = 1; i <= userNum; i++) {
                    factorial *= i;
                }

                System.out.println("\nThe factorial of " + userNum + " is " + factorial + ".");
                break;

            } catch (NumberFormatException e) {
                System.out.println();
                System.out.println("Please only enter a positive whole number.");
            }
        }

        scanner.close();
    }
}

File: day5/factorial.rs
use std::io;

fn main() {
    loop {
        println!("\nPlease enter a positive whole number (1 or greater):");

        let mut input = String::new();
        io::stdin().read_line(&mut input).expect("Failed to read input");

        match input.trim().parse::<u32>() {
            Ok(n) if n >= 1 => {
                let result = (1..=n as u128).product::<u128>();
                println!("The factorial of {} is {}.", n, result);
                break;
            }
            _ => println!("Please only enter a positive whole number."),
        }
    }
}
